addons/base_vat/models/res_partner.py

```diff
@@ -736,3 +736,28 @@ def write(self, values):
             country_id = values.get('country_id', self.country_id.id)
             values['vat'] = self._fix_vat_number(values['vat'], country_id)
         return super(ResPartner, self).write(values)
+
+    def check_vat_de(self, vat):
+        validate_vat_de = getattr(stdnum.util.get_cc_module('de', 'vat'), 'is_valid', None)
+        if validate_vat_de(vat) or self.validate_tin_de(vat):
```

Comment:
it seems like there is validation for Steuer-IdNr in stdnum: 

https://github.com/arthurdejong/python-stdnum/blob/1.13/stdnum/de/idnr.py

so this could probably be shortened to:

```suggestion
        validate_tin_de = getattr(stdnum.util.get_cc_module('de', 'idnr'), 'is_valid', None)
        if validate_vat_de(vat) or self.validate_tin_de(vat):
            return True
        return False
```
but the library allow characters: -./, and space inside the number, so if we really wanted only / we could still have your regex check